api/src/test/groovy/graphql/nadel/normalized/ExecutableNormalizedOperationToAstCompilerTest.groovy

```diff
@@ -0,0 +1,696 @@
+        def fields = createNormalizedFields(schema, query)
+        when:
+        def document = compileToDocument(schema, QUERY, null, fields).first
+        then:
+        AstPrinter.printAst(document) == '''query {
+  foo1(arg: "hello")
+  foo2(a: 123, b: true, c: 123.45)
+}
+'''
+    }
+
+    def "sets operation name"() {
+        def sdl = '''
+        type Query {
+            foo1(arg: String): String
+            foo2(a: Int,b: Boolean, c: Float): String
+        }
+        '''
+        def query = ''' {
+            foo1(arg: "hello")
+            foo2(a: 123, b: true, c: 123.45)
+        }
+        '''
+        GraphQLSchema schema = TestUtil.schema(sdl)
+        def fields = createNormalizedFields(schema, query)
+        when:
+        def document = compileToDocument(schema, QUERY, "My_Op23", fields).first
+        then:
+        AstPrinter.printAst(document) == '''query My_Op23 {
+  foo1(arg: "hello")
+  foo2(a: 123, b: true, c: 123.45)
+}
+'''
+    }
+
+    def "test input object arguments"() {
+        def sdl = '''
+        type Query {
+            foo1(arg: I): String
+        }
+        input I {
+            arg1: String
+            arg2: Int
+            arg3: ID
+            arg4: Boolean
+            arg5: Float
+            nested: I
+        }
+        '''
+        def query = '''{
+            foo1(arg: {
+             arg1: "Hello"
+             arg2: 123
+             arg3: "IDID"
+             arg4: false
+             arg5: 123.123
+             nested: {
+                 arg1: "Hello2"
+                 arg2: 1234
+                 arg3: "IDID1"
+                 arg4: null
+                 arg5: null
+             }
+            })
+        }
+        '''
+        GraphQLSchema schema = TestUtil.schema(sdl)
+        def fields = createNormalizedFields(schema, query)
+        when:
+        def document = compileToDocument(schema, QUERY, null, fields).first
+        then:
+        AstPrinter.printAst(document) == '''query {
+  foo1(arg: {arg1 : "Hello", arg2 : 123, arg3 : "IDID", arg4 : false, arg5 : 123.123, nested : {arg1 : "Hello2", arg2 : 1234, arg3 : "IDID1", arg4 : null, arg5 : null}})
+}
+'''
+    }
+
+    def "test mutations"() {
+        def sdl = '''
+        type Query {
+            foo1(arg: I): String
+        }
+        type Mutation {
+            foo1(arg: I): String
+        }
+        input I {
+            arg1: String
+        }
+        '''
+        def query = '''mutation {
+            foo1(arg: {
+             arg1: "Mutation"
+            })
+        }
+        '''
+        GraphQLSchema schema = TestUtil.schema(sdl)
+        def fields = createNormalizedFields(schema, query)
+        when:
+        def document = compileToDocument(schema, MUTATION, null, fields).first
+        then:
+        AstPrinter.printAst(document) == '''mutation {
+  foo1(arg: {arg1 : "Mutation"})
+}
+'''
+    }
+
+    def "test JSON when input is a variable"() {
```

Comment:
Can you move these to the bottom of the file and have like a separator comment? so the merge is easier in the future when we eventually move this code back to GraphQL Java.

e.g.

```
def "GraphQL Java tests above"() {
}

// --------------------------------------------------------------------------------------
// Custom JSON handling

def "test JSON when input is a variable"() {
}

…
```

Reply:
done